### PR TITLE
AF-1226: Improve error handling displaying additional info to the GenericErrorPopup

### PR DESCRIPTION
**Pull request by tiagobento**
Related:

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/770
https://github.com/errai/errai/pull/350
https://github.com/kiegroup/appformer/pull/404
https://github.com/kiegroup/kie-wb-common/pull/1925
https://github.com/kiegroup/drools-wb/pull/878
https://github.com/kiegroup/jbpm-wb/pull/1162
https://github.com/kiegroup/optaplanner-wb/pull/288
https://github.com/kiegroup/kie-wb-distributions/pull/773

### CODE REVIEW

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by jhrcek** [comment]
3 test failures in FDB are unrelated to the changes in the PR
```
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonProcessViolated[0:  JAXB KieServicesConfiguration{transport=REST,  serverUrl='http://localhost:39958/kie-server-services/services/rest/server'}] | 4.5 sec | 1
-- | -- | --
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonUserTaskViolated[1:  JSON KieServicesConfiguration{transport=REST,  serverUrl='http://localhost:39958/kie-server-services/services/rest/server'}] | 4.1 sec | 1
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonProcessViolated[4:  JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]

```

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/callbacks/DefaultErrorCallback.java [hunk 1]
```diff
@@ -23,7 +23,9 @@
 
 /**
  * Default Error handler for all Portable Exceptions
+ * @deprecated Please use GenericErrorPopup instead.
  */
+@Deprecated
```
Comment (jhrcek) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/callbacks/DefaultErrorCallback.java]: Just a question related to this deprecation: does it mean that there are still places in the workbench, that will display unexpected errors in "the old way"? Or will these be handled universally using the new GenericErrorPopup?

Comment (tiagobento) [reply uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/callbacks/DefaultErrorCallback.java]: Yes. This class is still used, but only by code actively wanting to do a custom error handling. I didn't dig in detail why, but the default error handling has changed, so not much to worry here.

**Comment by tiagobento** [comment]
@ederign Can we merge this and all the related PRs?

**Comment by ederign** [comment]
still need ack's in order to merge this.

File: pom.xml [hunk 2]
```diff
@@ -65,7 +65,7 @@
     <version.org.apache.sshd>1.6.0</version.org.apache.sshd>
     <!-- Custom Freemarker build with workaround for random concurrent access issue (annotation processors). See AF-600 for more info.-->
     <version.org.freemarker>2.3.26.jbossorg-1</version.org.freemarker>
-    <version.org.jboss.errai>4.2.2.Final</version.org.jboss.errai>
+    <version.org.jboss.errai>4.2.3-SNAPSHOT</version.org.jboss.errai>
```
Comment (ederign) [inline pom.xml]: [16:27:03]  ederign:	tbento can we move 7.7.x errai version to <version.org.jboss.errai>4.2.3-SNAPSHOT</version.org.jboss.errai> ?
[16:27:07]  ederign:	it's safe?
 tbento:	yes, that's the current version on errai's 4.2.x branch. but before we generate a release of the workbench we need to generate 4.2.3.Final

**Comment by tiagobento** [comment]
jenkins execute full downstream build

**Pull request merged**